surface/include/pcl/surface/impl/mls.hpp

```diff
@@ -738,10 +737,21 @@ pcl::MLSResult::computeMLSSurface (const pcl::PointCloud<PointT> &cloud,
   model_coefficients.head<3> ().matrix () = eigen_vector;
   model_coefficients[3] = -1 * model_coefficients.dot (xyz_centroid);
 
+  query_point = cloud.points[index].getVector3fMap ().template cast<double> ();
+
+  if (!std::isfinite(eigen_vector[0]) || !std::isfinite(eigen_vector[1]) || !std::isfinite(eigen_vector[2]))
+  {
+    // Invalid plane coefficients, this may happen if the input cloud is non-dense (it contains invalid points)
+    // or if the points in the neighborhood are linearly dependent (e.g. on a line).
+    // Keep the input point and stop here.
```

Comment:
I'm a little concerned over this statement. Even if the neighborhood points are linearly dependent and the matrix is not full rank, I don't see a reason for the eigenvectors to be populated with infinite values. It's been established in other issues that `eigen33` is buggy. Nevertheless, @ThorstenHarter did you observe the behavior of nan filled eigenvectors in a singular covariance matrix?

Reply:
@SergioRAgostinho Yes, I observed this if the points are on a line.
The matrix is then singular and vec1/vec2/vec3 resp. len1/len2/len3 in pcl::eigen33 are all zero.
The eigenvector is calculated as `eigenvector = vec1 / std::sqrt (len1);` and therefore nan.

Reply:
Thanks for the info.

I don't think it's a good idea to keep the comment on linear dependence around despite the fact that currently it is actually valid, because theoretically, it's incorrect.

Reply:
Okay, I have removed the comment in line 745.